### ISSUE TITLE
Custom weapon pictures stored in maps

### ISSUE THREAD

**Issue by 3da** [teeworlds/teeworlds#2490]
It would be useful for custom mods.

For example:
1. I add custom weapon picture, and specify weapon id.
2. Players play on the server.
3. Something happened.
4. Server sends that specific player has weapon with id.
5. Player switches weapon and see custom weapon picture (which is stored in map data)

**Comment by Dune-jr** [comment]
All these things would break compatibility and require 0.8, by the way.

**Comment by 3da** [comment]
Yeah, I know, but 0.8 will be somewhen. So it would be cool if 0.8 will have more opportunities for teeworlds server mod makers.

Mod makers prefer make mods only for server because default teeworlds client is already distributing with some channels.

And it would great if default teeworlds client will support more different mods, isn't it?

**Comment by 3da** [comment]
I am not sure that custom graphics and game entities should be transferred with map data. May be it should have another mechanism.

But the main purpose is to provide some new weapons (or some other objects) to clients from server without requirements to use different client version.

Many fine mods are dead because default teeworlds players cannot access mods with incompatible  protocol. But if protocol will be enough diverse it will not require to make custom teeworlds client.

**Comment by Dune-jr** [comment]
I guess this is covered by #2156 to some extent

**Comment by 3da** [comment]
> I guess this is covered by #2156 to some extent
That issue says nothing about protocol extensions. 🙂
